pkg/exporter_test.go

```diff
@@ -0,0 +1,252 @@
+	if m.err != nil {
+		return nil, m.err
+	}
+	return m.resources, nil
+}
+
+// mockCloudwatchClient implements the cloudwatch.Client interface
+type mockCloudwatchClient struct {
+	metrics           []*model.Metric
+	metricDataResults []cloudwatch.MetricDataResult
+	err               error
+}
+
+func (m *mockCloudwatchClient) ListMetrics(_ context.Context, _ string, _ *model.MetricConfig, _ bool, fn func(page []*model.Metric)) error {
+	if m.err != nil {
+		return m.err
+	}
+	if len(m.metrics) > 0 {
+		fn(m.metrics)
+	}
+	return nil
+}
+
+func (m *mockCloudwatchClient) GetMetricData(_ context.Context, _ []*model.CloudwatchData, _ string, _ time.Time, _ time.Time) []cloudwatch.MetricDataResult {
+	return m.metricDataResults
+}
+
+func (m *mockCloudwatchClient) GetMetricStatistics(_ context.Context, _ *slog.Logger, _ []model.Dimension, _ string, _ *model.MetricConfig) []*model.MetricStatisticsResult {
+	// Return a simple metric statistics result for testing
+	now := time.Now()
+	avg := 42.0
+	return []*model.MetricStatisticsResult{
+		{
+			Timestamp: &now,
+			Average:   &avg,
+		},
+	}
+}
+
+func TestUpdateMetrics_StaticJob(t *testing.T) {
+	ctx := context.Background()
+	logger := promslog.NewNopLogger()
+
+	// Create a simple static job configuration
+	jobsCfg := model.JobsConfig{
+		StaticJobs: []model.StaticJob{
+			{
+				Name:      "test-static-job",
+				Regions:   []string{"us-east-1"},
+				Roles:     []model.Role{{}},
+				Namespace: "AWS/EC2",
+				Dimensions: []model.Dimension{
+					{Name: "InstanceId", Value: "i-1234567890abcdef0"},
+				},
+				Metrics: []*model.MetricConfig{
+					{
+						Name:       "CPUUtilization",
+						Statistics: []string{"Average"},
+						Period:     300,
+						Length:     300,
+					},
+				},
+			},
+		},
+	}
+
+	factory := &mockFactory{
+		accountClient: mockAccountClient{
+			accountID:    "123456789012",
+			accountAlias: "test-account",
+		},
+		cloudwatchClient: mockCloudwatchClient{},
+	}
+
+	registry := prometheus.NewRegistry()
+
+	err := UpdateMetrics(ctx, logger, jobsCfg, registry, factory)
+	require.NoError(t, err)
+
+	// Verify the expected metric exists using testutil
+	expectedMetric := `
+		# HELP aws_ec2_cpuutilization_average Help is not implemented yet.
+		# TYPE aws_ec2_cpuutilization_average gauge
+		aws_ec2_cpuutilization_average{account_alias="test-account",account_id="123456789012",dimension_InstanceId="i-1234567890abcdef0",name="test-static-job",region="us-east-1"} 42
+	`
+
+	err = testutil.GatherAndCompare(registry, strings.NewReader(expectedMetric), "aws_ec2_cpuutilization_average")
+	require.NoError(t, err, "Metric aws_ec2_cpuutilization_average should match expected output")
+}
+
+func TestUpdateMetrics_DiscoveryJob(t *testing.T) {
+	ctx := context.Background()
+	logger := promslog.NewNopLogger()
+
+	// Create a ScrapeConf configuration and convert it to model.JobsConfig
+	conf := &config.ScrapeConf{
```

Comment:
What do you think about creating `model.JobsConfig` instead of using the `ToModelConfig` method and making it from the `ScrapeConf`?

In this case, no changes for the config ScrapeConf object are required.